### ISSUE TITLE
Lizard groups nearly 100 lines into one "function" in a Rust file

### ISSUE THREAD

**Issue by rjray** [terryyin/lizard#363]
This is the file, at the current version: https://github.com/rjray/mscs-thesis-project/blob/811b319cd81d7901a248a5cebe840daeb63a2a75/src/Rust/aho_corasick/src/main.rs
This should remain good even if I make further changes to the file.
When I run the command `lizard Rust/aho_corasick/src/main.rs`, the resulting analysis counts lines 50-147 as a single function it calls `iter`. This starts inside an `impl` block for one struct, and completely subsumes a second struct/impl definition as well as two other functions.

**Comment by 503440692** [comment]
![image](https://user-images.githubusercontent.com/66709975/202461274-c1a372b5-b2a7-40c3-aee8-ed6e9cdb47eb.png)

There are syntax error in your source code.

**Comment by rjray** [comment]
That is incorrect. The code as written is correct, this is valid Rust syntax. The change you suggest would actually break the code.

**Comment by a1phyr** [comment]
Is there any progress on this ?

**Comment by rjray** [comment]
Bumping for visibility. I used this tool for my MSCS thesis research last fall, but had to use a separate tool for CCN in Rust. I am continuing this research, and would really like to be able to use the same tool (this tool) for all languages in my study (C, C++, Rust, possibly Go and Java as well). Switching tools for one or two of the languages makes it harder to compare the CCN of the different languages.
